lib/Interpreter/Interpreter.cpp

```diff
@@ -356,84 +358,84 @@ namespace cling {
-      // #define __dso_handle is used to mitigate the following problems:
-      //  1. Type of __dso_handle is void* making assignemnt to it legal
-      //  2. Making it void* const in cling would mean possible type mismatch
-      //  3. Cannot override void* __dso_handle in child Interpreter
-      //  4. On Unix where the symbol actually exists, __dso_handle will be
-      //     linked into the code before the JIT can say otherwise, so:
-      //      [cling] __dso_handle // codegened __dso_handle always printed
-      //      [cling] __cxa_atexit(f, 0, __dso_handle) // seg-fault
-      //  5. Code that actually uses __dso_handle will fail as a declaration is
-      //     needed which is not possible with the macro.
-      //  6. Assuming 4 is sorted out in user code, calling __cxa_atexit through
-      //     atexit below isn't linking to the __dso_handle symbol.
-
-      Strm << "#define __dso_handle ((void*)" << thisP << ")\n";
-
-      // Use __cxa_atexit to intercept all of the following routines
-      Strm << Linkage << " int __cxa_atexit(void (*f)(void*), void*, void*) "
-           << cxa_atexit_is_noexcept << ";\n";
-
-      // C atexit, std::atexit
-      Strm << Linkage << " int atexit(void(*f)()) " << Attr << " { return "
-                        "__cxa_atexit((void(*)(void*))f, 0, __dso_handle); }\n";
-      Globals.push_back("atexit");
-
-      // C++ 11 at_quick_exit, std::at_quick_exit
-      if (LangOpts.CPlusPlus && LangOpts.CPlusPlus11) {
-        Strm << LinkageCxx << " int at_quick_exit(void(*f)()) " << Attr <<
-              " { return __cxa_atexit((void(*)(void*))f, 0, __dso_handle); }\n";
-        Globals.push_back("at_quick_exit");
-      }
+    // While __dso_handle is still overriden in the JIT below,
+    // #define __dso_handle is used to mitigate the following problems:
+    //  1. Type of __dso_handle is void* making assignemnt to it legal
+    //  2. Making it void* const in cling would mean possible type mismatch
+    //  3. Cannot override void* __dso_handle in child Interpreter
+    //  4. On Unix where the symbol actually exists, __dso_handle will be
+    //     linked into the code before the JIT can say otherwise, so:
+    //      [cling] __dso_handle // codegened __dso_handle always printed
+    //      [cling] __cxa_atexit(f, 0, __dso_handle) // seg-fault
+    //  5. Code that actually uses __dso_handle will fail as a declaration is
+    //     needed which is not possible with the macro.
+    //  6. Assuming 4 is sorted out in user code, calling __cxa_atexit through
+    //     atexit below isn't linking to the __dso_handle symbol.
+
+    Strm << "#define __dso_handle ((void*)" << ThisP << ")\n";
+
+    // Use __cxa_atexit to intercept all of the following routines
+    Strm << Linkage << " int __cxa_atexit(void (*f)(void*), void*, void*) "
+         << cxa_atexit_is_noexcept << ";\n";
+
+    // C atexit, std::atexit
+    Strm << Linkage << " int atexit(void(*f)()) " << Attr << " { return "
```

Comment:
In that case, neither --noruntime nor -fsyntax-only will work for rootcling's PCH generation, so that's not an option. I don't want to throw in a third flag.

Please select whether you want to suppress these definitions (all that can trigger duplicate definition errors, i.e. function and variable definitions as well as preprocessor macro definitions) for --noruntime or -fsyntax-only. For me it makes more sense to suppress them for -fsyntax-only, given that we know we won't have static destruction as we won't have any codegen.